internal/state/state.go

```diff
@@ -143,6 +144,14 @@ func New(path util.AbsolutePath, accountName, configName, targetName string, sav
 			return nil, err
 		}
 	}
+
+	// Check that the secrets passed are in the config
+	for secret := range secrets {
+		if !cfg.HasSecret(secret) {
```

Comment:
This double loops since `cfg.HasSecret` does a loop over the Secrets defined in the Configuration, but that felt acceptable since:
- these will most likely be relatively small
- iterating through everything gives us a good error showing the first secret not in the config

Based on feedback I can expanded this to include an error of _all_ secrets passed that were not in the configuration if we think that is helpful. I wrote this as fail fast first.